src/src/DataStructs/PluginStats.cpp

```diff
@@ -156,80 +167,64 @@ bool PluginStats::plugin_get_config_value_base(struct EventStruct *event, String
-    if (equals(command, F("avg"))) { // [taskname#valuename.avg] Average value of the last N kept samples
-      value   = getSampleAvg();
-      success = true;
+  } else if (matchedCommand(command, F("avg"), nrSamples)) {
+    success = nrSamples != 0;
+
+    if (nrSamples < 0) { // [taskname#valuename.avg] Average value of the last N kept samples
+      value = getSampleAvg();
     } else {
       // Check for "avgN", where N is the number of most recent samples to use.
-      int nrSamples = 0;
-
-      if (validIntFromString(command.substring(3), nrSamples)) {
-        if (nrSamples > 0) {
-          // [taskname#valuename.avgN] Average over N most recent samples
-          value   = getSampleAvg(nrSamples);
-          success = true;
-        }
+      if (nrSamples > 0) {
+        // [taskname#valuename.avgN] Average over N most recent samples
+        value = getSampleAvg(nrSamples);
       }
     }
-  } else if (command.startsWith(F("stddev"))) {
-    if (equals(command, F("stddev"))) { // [taskname#valuename.stddev] Std deviation of the last N kept samples
-      value   = getSampleStdDev();
-      success = true;
+  } else if (matchedCommand(command, F("stddev"), nrSamples)) {
+    success = nrSamples != 0;
+
+    if (nrSamples < 0) { // [taskname#valuename.stddev] Std deviation of the last N kept samples
+      value = getSampleStdDev();
     } else {
       // Check for "stddevN", where N is the number of most recent samples to use.
-      int nrSamples = 0;
-
-      if (validIntFromString(command.substring(6), nrSamples)) {
-        if (nrSamples > 0) {
-          // [taskname#valuename.stddevN] Std. deviation over N most recent samples
-          value   = getSampleStdDev(nrSamples);
-          success = true;
-        }
+      if (nrSamples > 0) {
+        // [taskname#valuename.stddevN] Std. deviation over N most recent samples
+        value = getSampleStdDev(nrSamples);
       }
     }
-  } else if (equals(command, F("size"))) { // [taskname#valuename.size] Number of saved samples
+  } else if (matchedCommand(command, F("size"), nrSamples)) {
+    // [taskname#valuename.size] Number of saved samples
     value   = _samples.size();
     success = true;
-  } else if (command.startsWith(F("sample"))) {
-    if (equals(command, F("sample"))) { // [taskname#valuename.sample] The first (oldest) sample saved.
-      value   = _samples[0];
-      success = true;
-    } else {
-      int nrSamples = 0;
+  } else if (matchedCommand(command, F("sample"), nrSamples)) {
+    success = nrSamples != 0;
 
-      if (validIntFromString(command.substring(6), nrSamples)) {
-        if (nrSamples > 0) {            // [taskname#valuename.sampleN] Sample N (1 - last (current), N>[number of examples] - return error value)
-          value   = getSample(nrSamples);
-          success = true;
-        }
+    if (nrSamples < 0) {   // [taskname#valuename.sample] The last sample saved.
+      value = getSample(0);
```

Comment:
value = getSample(_samples.size()) or value = _samples[0]

Reply:
See my other comment.
And these kinds of comments from you are the reason I did not yet merge it :)